Automattic/jetpack
# Tiled Gallery block: Adding styles to squareish tiled gallery images on WoA sites

Fixes #27614


#### Changes proposed in this Pull Request:

* This PR adds a check to see if a site is a WoA or localhost/local dev site, then adds a stylesheet which applies an equal aspect ratio to the relevant squareish images within Tiled Galleries.
* It targets all WoA sites rather than private WoA sites because the issue remains on private sites that have become public (unless all posts / pages in question are re-saved, which could be a significant amount).

* Other solutions that have been approached included adding a JS file in the main TG block directory and importing that / calling the relevant function from edit.js and view.js (there appears to be an issue with the TG view.js file on Atomic sites generally in this case which needs more investigation), enqueueing styles and scripts from within the main Tiled Gallery block PHP file (none of which I've managed to enqueue so far unless the file lives outside the block directory), and echoing a `<script>` within the PHP file directly - this does work but the build action fails - `curl: (3) URL using bad/illegal format or missing URL. Error: Process completed with exit code 3.`. The current solution works but isn't the most elegant. Putting this up for review now though so there is at least an interim solution before more investigation can take place.

Before:
<img width="400" alt="Screen Shot 2022-12-09 at 2 17 27 pm" src="https://user-images.githubusercontent.com/16754605/206722182-8dfff9df-6a49-41e7-a4df-b564bde8f9a0.png">


After:

<img width="400" alt="Screen Shot 2022-12-09 at 2 16 55 pm" src="https://user-images.githubusercontent.com/16754605/206722234-3133e118-5b1e-4a80-a4a5-cb73c258d03b.png">

#### Other information:


#### Jetpack product discussion

pdWQjU-av-p2 

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* First try replicating the issue with square / circular tiled gallery images on either a localhost - local dev site or a private WoA site.
* To do this, create a new post / page on that site and add a Tiled Gallery block with images that don't have equal aspect ratios (no squares or circles). Choose 'circles' or 'square tiles' from the block styles, and you should notice the circles are ovals and the squares are rectangular. This should be the same on the front-end as well.
* Then apply this PR using the Jetpack beta tester plugin on a WoA site or checking it out directly within a local development environment.
* The correct (equal) aspect ratios should now be visible for those images - circles and squares.
